### PR TITLE
mute should support duration

### PR DESCRIPTION
**Pull request by ElvinEfendi**
fixes https://github.com/Shopify/doggy/issues/30

@Shopify/traffic

### CODE REVIEW

File: lib/doggy/duration.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+require 'active_support/core_ext/module'
+require 'active_support/core_ext/integer'
+
+class Duration
+  DURATION_FORMAT = /
+    \A
```
Comment (klautcomputing) [inline lib/doggy/duration.rb]: Do you need newlines? `\A .. \z` vs `$ .. ^`

File: lib/doggy/duration.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      "use digits followed by a unit (#{DURATION_UNITS.keys.join(', ')} for #{DURATION_UNITS.values.join(', ')})"
+    end
+    duration = DURATION_UNITS.values.inject(0) do |as_duration, unit|
+      as_duration + match[unit].to_i.public_send(unit)
+    end
+    new(duration)
+  end
+
+  def initialize(duration)
+    @as_duration = if duration.is_a?(ActiveSupport::Duration)
```
Comment (klautcomputing) [inline lib/doggy/duration.rb]: would look much cleaner in 2 lines 
```
as_duration = duration if if duration.is_a?(ActiveSupport::Duration)
as_duration = duration.to_i.seconds if !duration.is_a?(ActiveSupport::Duration)
```
or just move the `@as_duration` inside the if
``` 
if duration.is_a?(ActiveSupport::Duration)
  @as_duration = ...
else
  ...
end
```

File: lib/doggy/duration.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    's' => :seconds,
+    'm' => :minutes,
+    'h' => :hours,
+    'd' => :days,
+  }
+
+  def self.parse(value)
+    unless match = DURATION_FORMAT.match(value)
+      raise ArgumentError, "not a duration: #{value.inspect}, "\
+      "use digits followed by a unit (#{DURATION_UNITS.keys.join(', ')} for #{DURATION_UNITS.values.join(', ')})"
```
Comment (klautcomputing) [inline lib/doggy/duration.rb]: how something like `#{DURATION_UNITS.map { |k, v| k + "for " v}.join(', ')`

**Comment by ElvinEfendi** [comment]
The duration class has been copied from: https://github.com/Shopify/spy/blob/master/lib/spy/duration.rb

FYI, @klautcomputing

File: lib/doggy/duration.rb (hunk 1 above)
Comment (ElvinEfendi) [inline lib/doggy/duration.rb]: imo, it is always safest to use `\A .. \z`

**Pull request merged**